helios-testing/src/main/java/com/spotify/helios/testing/HeliosDeploymentResource.java

```diff
@@ -97,12 +98,32 @@ public Boolean call() throws Exception {
           return null;
         }
 
-        if (!hosts.isEmpty()) {
-          log.info("Ensured that at least one agent is available in this HeliosDeployment, "
+        if (hosts.isEmpty()) {
+          log.debug("0 agents in {}, will retry", deployment);
+          return null;
+        }
+
+        log.info("Ensured that at least one agent has registered with ZooKeeper in this"
+                 + "HeliosDeployment. Now checking if it is UP.");
+
+        // Check that at least one host is UP (is maintaining a reasonably reliable
+        // connection to ZK) in addition to registering.
+        final ListenableFuture<HostStatus> statusFuture = client.hostStatus(hosts.get(0));
```

Comment:
nitpick: Is there any benefit to using `hostStatus(String)` over `hostStatus(List<String>)`? this doesn't matter i just personally don't like `.get(0)` if it can be avoided :)